File: DSA Implementations/Python/linear_search_matrix.py
def linear_search(arr,k):
    for i in range(len(arr)):
        for j in range(len(arr[i])):
            if arr[i][j] == k:
                return (i,j)
    return -1

rows = int(input("Enter the number of rows in the array : "))
cols = int(input("Enter the number of columns in the array : "))
arr = []

for i in range(rows):
    row=[]
    for j in range(cols):
        row.append(int(input(f"Enter element ({i},{j}) : ")))
    arr.append(row)

k =  int(input("Enter the element to be searched : "))

linear_search_result = linear_search(arr, k)

if linear_search_result != -1:
    print(f"Element {k} found at index {linear_search_result}.")
else:
    print(f"Element {k} not found in the array.")

File: DSA Implementations/C++/linear_search_matrix.cpp
#include <iostream>
#include <vector>
using namespace std;

bool linear_search(const vector<vector<int>> &arr, int k)
{
    for (int i = 0; i < arr.size(); ++i)
    {
        for (int j = 0; j < arr[i].size(); ++j)
        {
            if (arr[i][j] == k)
            {
                cout << "Element found at index (" << i << ", " << j << ")" << endl;
                return true;
            }
        }
    }
    return false;
}

int main()
{
    int rows, cols, k;

    cout << "Enter the number of rows in the matrix: ";
    cin >> rows;

    cout << "Enter the number of columns in the matrix: ";
    cin >> cols;

    vector<vector<int>> arr(rows, vector<int>(cols));

    cout << "Enter the elements of the matrix: " << endl;
    for (int i = 0; i < rows; ++i)
    {
        for (int j = 0; j < cols; ++j)
        {
            cin >> arr[i][j];
        }
    }

    cout << "Enter the element to be searched: ";
    cin >> k;

    bool is_found = linear_search(arr, k);

    if (!is_found)
    {
        cout << "Element not found in the matrix" << endl;
        return 1;
    }
    else
    {
        return 0;
    }
}